packages/data/abstract/pollbydate/README.md

```diff
@@ -0,0 +1,61 @@
+# Poll by Date data
+
+Is an abstract data module which have:
+- `PollByDate` entity
+- `PollByDateRepository` data controller
+
+It is needed to correctly save one user poll per date which is used as a dt integer key.
+
+Why not use a Map itself:
+- Provider package and inherited widgets in Flutter work best with a concrete instances of a concrete objects in a search tree.
+- Abstraction.
+- DRY, less code for serialziation, it is written once.
+
+## Entity:
+
+`class PollByDate` with only one field
+- [`Map<int, UserPoll>`](https://github.com/rimmer/my_prophet/tree/master/packages/data/abstract/userpoll/README.md) date
+
+Constructor:
+```
+PollByDate({
+    Map<int, UserPoll> map
+}) : date = map ?? {};
+```
+
+## Data Controller:
+
+Reposiory pattern.
+
+---
+
+`abstract class PollByDateRepository`
```

Comment:
I would imagine it more like one repo (`PollRepository`) with various ways of retrieving data.
And 'by date' would be default way of getting it.